qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Bravyi-Kitaev Super Fast Mapper."""
+
+from enum import Enum
+from typing import List, Tuple
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import Pauli, SparsePauliOp
+from qiskit_nature.operators.second_quantization import FermionicOp
+from .fermionic_mapper import FermionicMapper
+
+
+class BravyiKitaevSuperFastMapper(FermionicMapper):
+    """The Bravyi-Kitaev super fast fermion-to-qubit mapping.
+
+    This implementation follows closely Reference [1].
+
+       References:
+           [1]: Bravyi-Kitaev Superfast simulation of electronic structure on a quantum computer
+                Kanav Setia and James D Whitfield
+                JCP Volume 148, Issue 16 - Published April 2018
+                arXiv:1712.00446
+    """
+
+    def map(self, second_q_op: FermionicOp) -> PauliSumOp:
+        if not isinstance(second_q_op, FermionicOp):
+            raise TypeError("Type ", type(second_q_op), " not supported.")
+
+        edge_list = bksf_edge_list_fermionic_op(second_q_op)
+        sparse_pauli = _convert_operators(second_q_op, edge_list)
+
+        ## Simplify and sort the result
+        sparse_pauli = sparse_pauli.simplify()
+        indices = sparse_pauli.table.argsort()
+        table = sparse_pauli.table[indices]
+        coeffs = sparse_pauli.coeffs[indices]
+        sorted_sparse_pauli = SparsePauliOp(table, coeffs)
+
+        return PauliSumOp(sorted_sparse_pauli)
+
+
+class TermType(Enum):
+    """Denotes the type of interaction of a Fermionic operator.
+    Allowed types of interaction are: NUMBER, EXCITATION, COULOMB_EXCHANGE, NUMBER_EXCITATION,
+    DOUBLE_EXCITATION.
```

Comment:
I agree with @mrossinek here. Also I think is counter-intuitive the following enumeration on the type of terms.

Reply:
I did not intend that documentation for these interaction term types should be exposed to users. I am not sure how to control what documentation is automatically included in API documentation. But, I do want the term types and their relation to the literature to be very clear when reading or working on the code. I think all the repetition of the types in doc strings was a hold-over from when I used strings rather than an enum.

Perhaps repeating the types of the Enum in the doc string is not useful. It would be more informative to have something like: "Denotes the type of interaction of a Fermionic operator. The names of the types follow the usage of Reference [1] above". If the reader reads `TermType` in the methods below, she can look at the definition of `TermType` to find the meaning.

> I think is counter-intuitive the following enumeration on the type of terms.

I'm not sure what this means. Do you mean it's better not to explicitly number them?

Reply:
I'm not sure I fully understand the comments from @mrossinek and @pbark. But, I have removed all documentation of the allowed values/labels of the enum from the file. The reasoning is that the enum is not exposed in any API and is only used in the file. The list of labels is self documenting. We use type annotations, so whereever `TermType` appears, one can check the definition of the enum to get the allowed values.

Reply:
What I meant was that I did not like the duplication of the Enum's contents in its own docstring. At least not in that format because it did not really add any information/explanation about the different types.